modules/service/src/main/scala/lucuma/odb/service/GroupService.scala

```diff
@@ -249,6 +320,79 @@ object GroupService {
          WHERE
            c_group_id = $group_id
        """.query(program_id)
+
+      /** Select group elements in a group, in order. */
+      val SelectGroupElements: Query[Group.Id, GroupElement.Id] =
+        sql"""
+          SELECT c_group_id, c_observation_id
+          FROM (
+            SELECT null c_group_id, c_observation_id, c_group_index
+            FROM t_observation WHERE c_group_id = $group_id
+            UNION
+            SELECT c_group_id, null, c_parent_index
+            FROM t_group WHERE c_parent_id = $group_id
+          ) sub ORDER BY c_group_index
+        """
+          .contramap[Group.Id](a => (a, a))
+          .query(
+            (group_id.opt ~ observation_id.opt).emap:
+              case (None, Some(oid)) => Right(Right(oid))
+              case (Some(gid), None) => Right(Left(gid))
+              case (a, b)            => Left("SelectGroupElements: unpossible row: $a, $b")
+          )
+
+      /** Select a `CreateGroupInput` for a given `Group.Id` that can be used to create an empty clone. */
+      val SelectGroupAsInput: Query[Group.Id, (CreateGroupInput, Boolean)] =
```

Comment:
Should this filter out system groups?

Reply:
It selects the system bit and delivers an error if it's true. I could have filtered it out but that would also mask the case where the group id is invalid. Not sure if we need an explicit error in that case.